packages/ast-spec/src/declaration/TSDeclareFunction/spec.ts

```diff
@@ -1,11 +1,35 @@
 import type { AST_NODE_TYPES } from '../../ast-node-types';
 import type { FunctionBase } from '../../base/FunctionBase';
-import type { BlockStatement } from '../../statement/BlockStatement/spec';
 
-// TODO(#1852) - async + declare are semantically invalid together
-export interface TSDeclareFunction extends FunctionBase {
+interface TSDeclareFunctionBase extends FunctionBase {
   type: AST_NODE_TYPES.TSDeclareFunction;
-  body: BlockStatement | undefined;
+  /**
+   * TS1183: An implementation cannot be declared in ambient contexts.
+   */
+  body: undefined;
```

Comment:
Tightened: I think this is safe?

Reply:
This is not correct to do.
The error you cited is a **semantic** error, not a **syntax** error.
This means that TS produces an AST for the error and us we do too.

[playground](https://typescript-eslint.io/play/#ts=5.4.5&showAST=es&fileType=.tsx&code=FAEwpgxgNghgTmABAMwK4DsIBcCWB7dFPPACgEpEBvYRRCAgZy0QA9EBeRARgG5gBfYEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)